### ISSUE TITLE
How can I escape the ***client.Run(cmd)*** loop?

### ISSUE THREAD

**Issue by bashery** [melbahja/goph#32]
Question: How can I escape the client.Run(cmd) loop?
When I call client.Run(cmd) or something my program can't go to the next tasks.
I am trying to run the command in the background of the remote host. Close the connection and allow the app to do other things.
Currently, I forcibly close the program like this.
```go
       go func() {
           _, _:= sshClient.Run(myCmd + "&") // Infinite loop.
           sshClient.Close() // The app cannot execute this line 
       }()
```

I'm looking for a logical solution to my problem.
thank you so much

**Comment by melbahja** [comment]
Hi, 

go routines is a good way to do it, without "&" at the end of cmd.

**Comment by bashery** [comment]
Yes. This is a temporary solution. The remote application will remain running as long as my application is running. When my app stops the remote app will stop.
I'm still looking for a good solution to the problem.
Another solution is to use SystemD
It is a good solution. But I avoid it for technical reasons.

**Comment by george-zakharov** [comment]
> Yes. This is a temporary solution. The remote application will remain running as long as my application is running. When my app stops the remote app will stop. I'm still looking for a good solution to the problem. Another solution is to use SystemD It is a good solution. But I avoid it for technical reasons.

Hi!
Did you find any good solution of this problem?

**Comment by melbahja** [comment]
I think that `nohup` it's gonna help you.

**Comment by george-zakharov** [comment]
> I think that `nohup` it's gonna help you.
Thank you for this idea.
I'll check it on my case.

**Comment by george-zakharov** [comment]
> I think that `nohup` it's gonna help you.
I've checked.
Not working in my case.
App still can't go to the next step after `Run()` method.
Looking for another solution.
If you have some more advices, I will be pleased to check them :)

**Comment by melbahja** [comment]
You can use `Start()` returns immediately without blocking:

```go
cmd, _ := client.Command("sleep", "60")
cmd.Start() // command keeps running
// ... do other things ...
```

For background commands that survive session close, `nohup` should work:

```go
client.Run("nohup some-long-task > /dev/null 2>&1 &")
```

Closing! 🙏
